viewing_party/main.py

```diff
@@ -0,0 +1,135 @@
+
+
+def create_movie(title, genre, rating):
+    movie_dictionary = {
+        "title": title, 
+        "genre": genre, 
+        "rating": rating
+        } 
+    if not title == None and not genre == None and not rating == None:
+        return movie_dictionary
+    else:
+        return None 
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie) 
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    list_of_watched_movies = user_data["watched"]
+    watch_list = user_data["watchlist"]
+    
+    for i in range (len(watch_list)):
+        title_of_watchlist_movie = user_data["watchlist"][i]["title"]
+        if title == title_of_watchlist_movie:
+            list_of_watched_movies.append(watch_list.pop(i))
+    return user_data
+    
+def get_watched_avg_rating(user_data):  
+    total = 0
+    average = 0.0
+    if len(user_data["watched"]) == 0:
+        average = 0.0
+    else:
+        for movie in user_data["watched"]:
+            total += movie["rating"]
+        average = total / len(user_data["watched"])
+    return average
+
+def get_most_watched_genre(user_data):
+    list_of_movies = user_data["watched"]
+    
+    if not list_of_movies:
+        return None
+    
+    else:
+        list_of_genres = [] 
+        count = 0
+        most_pop_genre = ""
+        
+        for movie in list_of_movies:
+            list_of_genres.append(movie["genre"])
+            for genre in list_of_genres:
+                frequency = list_of_genres.count(genre)
+                if frequency > count:
+                    count = frequency
+                    most_pop_genre = genre
```

Comment:
Ohh this is an interesting way of getting the most_watched_genre. Another way we could have solved this is by using a frequency map (aka dictionary).